tests/CRDTProperties.hs

```diff
@@ -0,0 +1,323 @@
+-- |   module:    CRDTProperties
+--     copyright: (c) 2016 Sentenai
+--     author:    Antonio Nikishaev <me@lelf.lu>
+--     license:   Apache
+-- 
+{-# LANGUAGE OverloadedStrings, FlexibleContexts, TupleSections, ScopedTypeVariables,
+    GADTs, StandaloneDeriving, UndecidableInstances, PatternGuards, MultiParamTypeClasses #-}
+{-# OPTIONS_GHC -fno-warn-orphans #-}
```

Comment:
Using `UndecidableInstances` worries me a bit here, I don't like using it if it can be helped.
Seems to only be for this line:
```haskell
deriving instance (Show op, C.CRDT a o) => Show (Op a op)
```